include/cupla/device/math/Common.hpp

```diff
@@ -62,60 +66,59 @@ namespace detail
             >::value,
             "The math concept implementation for the type 'Acc' and 'AccThreadSeq' must be equal"
         );
+#endif
         return AccMathConcept{};
     }
 } // namespace detail
 
-/* Using the free alpaka functions `alpaka::math::*` will result into `__host__ __device__`
- * errors, therefore the alpaka math trait must be used.
- */
-#define CUPLA_UNARY_MATH_FN(functionName, alpakaMathConcept, alpakaMathTrait)  \
+#define CUPLA_UNARY_MATH_FN_DETAIL(accOrMathImpl, functionName, alpakaMathConcept, alpakaMathTrait)  \
     /**                                                                        \
      * @tparam T_Type argument type                                            \
      * @param arg input argument                                               \
      */                                                                        \
     ALPAKA_NO_HOST_ACC_WARNING                                                 \
     template< typename T_Type >                                                \
-    ALPAKA_FN_ACC ALPAKA_FN_INLINE auto functionName(                          \
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE auto functionName(                     \
         T_Type const & arg                                                     \
     )                                                                          \
-    /* return type is required because nvcc can not detect the return type     \
-     * for device functions.                                                   \
+    /* return type is required for the compiler to detect host, device         \
+     * function quallifier correctly                                           \
```

Comment:
typo, double `l` in `qualifier`. It was also copied to another similar comment below.